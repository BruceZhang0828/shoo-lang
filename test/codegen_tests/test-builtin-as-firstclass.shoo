

function foo(func(string;void) bar, func(int;string) baz) void {
  bar("hello world" + baz(1234));
  return;
}

foo(println, str_of_int);
