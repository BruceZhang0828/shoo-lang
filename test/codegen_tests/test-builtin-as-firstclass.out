hello world1234
5
